pennylane/qnode.py

```diff
@@ -878,10 +904,15 @@ def to_torch(self, dtype=None):
 
             if self.interface != "torch" and self.interface is not None:
                 # Since the interface is changing, need to re-validate the tape class.
+                # if method was changed from "best", set it back to best
+                if self.diff_method_change:
+                    diff_method = "best"
+                else:
+                    diff_method = self.diff_method
                 self._tape, interface, self.device, diff_options = self.get_tape(
-                    self._original_device, "torch", self.diff_method
+                    self._original_device, "torch", diff_method
                 )
-
+                self.set_best_diff_method(diff_options)

```

Comment:
These could be moved inside the `if self.diff_method_change:`. No need to set it differently unless it was changed before, right?

Reply:
Yes. I have used it like this.